### PR TITLE
closes #631, adding the support to block users

### LINKED ISSUE #631: Add the ability to block a user
Hello, this is a feature request for adding an argument `is_blocked` (boolean, default `false`) to the `gitlab_user` resource. Thanks for your consideration.

### PR DESCRIPTION
**Pull request by hallaj**
Just adding a basic implementation for blocking user in the terraform provider

closes #631

### CODE REVIEW

File: gitlab/resource_gitlab_user.go [hunk 1]
```diff
@@ -182,6 +188,22 @@ func resourceGitlabUserUpdate(d *schema.ResourceData, meta interface{}) error {
...
 	}
 
+	if d.HasChange("state") {
+		if d.Get("state") == "active" {
+			err := client.Users.UnblockUser(id)
+
+			if err != nil {
+				return err
+			}
+		} else if d.Get("state") == "deactivated" {
```
Comment (armsnyder) [inline gitlab/resource_gitlab_user.go:198]: This should be `blocked` not `deactivated` right?

File: gitlab/resource_gitlab_user_test.go [hunk 2]
```diff
@@ -146,6 +147,7 @@ type testAccGitlabUserExpectedAttributes struct {
 	SkipConfirmation bool
 	External         bool
 	Note             string
+	State            string
 }
 
 func testAccCheckGitlabUserAttributes(user *gitlab.User, want *testAccGitlabUserExpectedAttributes) resource.TestCheckFunc {
```
Comment (armsnyder) [inline gitlab/resource_gitlab_user_test.go:153]: You'll need to update this function to check the new attribute

**Review by armsnyder — CHANGES_REQUESTED** [review]
Thanks for the PR! 😄 Could you rebase please?

We've made some significant changes recently by moving to the Terraform SDK V2. You'll also need to add a `Description` field to the new attribute, and then run `make reviewable` to generate documentation.

See the CONTRIBUTING.md for more up to date info.

File: gitlab/resource_gitlab_user.go (hunk 1 above)
Comment (hallaj) [reply gitlab/resource_gitlab_user.go]: Thanks for catching that, fixed.

File: gitlab/resource_gitlab_user_test.go (hunk 2 above)
Comment (hallaj) [reply gitlab/resource_gitlab_user_test.go:255]: Created a new one instead below to cover the blocked use case. Hope that's fine?

File: gitlab/resource_gitlab_user.go [hunk 3]
```diff
@@ -105,6 +111,7 @@ func resourceGitlabUserSetToState(d *schema.ResourceData, user *gitlab.User) {
 	d.Set("is_admin", user.IsAdmin)
 	d.Set("is_external", user.External)
 	d.Set("note", user.Note)
+	d.Set("state", user.State)
 }
 
 func resourceGitlabUserCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (armsnyder) [inline gitlab/resource_gitlab_user.go:117]: Can you add this to the "create" function _after_ the call to `d.SetId` --

```go
if d.Get("state") == "blocked" {
	if err := client.Users.BlockUser(id); err != nil {
		return diag.FromErr(err)
	}
}
```

File: gitlab/resource_gitlab_user.go [hunk 4]
```diff
@@ -92,6 +92,12 @@ func resourceGitlabUser() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 			},
+			"state": {
+				Description: "String, defaults to 'active'. The state of the user account.",
```
Comment (armsnyder) [inline gitlab/resource_gitlab_user.go:96]: The description should mention "blocked" as well. You could also add a field here - `ValidateFunc: validation.StringInSlice([]string{"active", "blocked"}, false)` - to validate values.

File: gitlab/resource_gitlab_user_test.go (hunk 2 above)
Comment (armsnyder) [reply gitlab/resource_gitlab_user_test.go:255]: Looks good!

**Review by armsnyder — CHANGES_REQUESTED** [review]

File: gitlab/resource_gitlab_user.go (hunk 3 above)
Comment (hallaj) [reply gitlab/resource_gitlab_user.go]: Added

File: gitlab/resource_gitlab_user.go (hunk 4 above)
Comment (hallaj) [reply gitlab/resource_gitlab_user.go]: Added both

**Review by armsnyder — APPROVED** [review]
lgtm, thanks!

**Pull request merged**